dashboard/views.py

```diff
@@ -350,12 +350,17 @@ def grade_distribution(request, course_id=0):
     if df.empty or df['current_grade'].isnull().all():
         return HttpResponse(json.dumps({}), content_type='application/json')
 
-    df['tot_students'] = df.shape[0]
+    result = {}
+    result['summary'] = {}
+    result['summary']['current_user_grade'] = df['current_user_grade'].values[0]
```

Comment:
using the double array can be confusing when somebody is reading the code. Can this be simplified like this?

```grades = [23,34,45,55,20,10]
summary = dict()
summary['tot_stu']= 80
summary['median']=30
sum = {}
sum['summary']=summary
sum['grades']=grades
print(sum)
print(json.dumps(sum))

Result : {"summary": {"tot_stu": 80, "median": 30}, "grades": [23, 34, 45, 55, 20, 10]}
```

Reply:
I would agree about the double array, though having `sum` and `summary` is confusing to me. Like @pushyamig is suggesting, try first building the summary dictionary and then assign it all at once to a key in the parent dictionary. I would probably name the outer dictionary something like `grades_dist_data` and the summary dictionary something like `grades_summary`.